storage/innobase/xtrabackup/src/xtrabackup.cc

```diff
@@ -4074,8 +4074,9 @@ void xtrabackup_backup_func(void) {
     strcpy(metadata_type_str, "incremental");
     metadata_from_lsn = incremental_lsn;
   }
-  metadata_to_lsn = redo_mgr.get_last_checkpoint_lsn();
-  metadata_last_lsn = redo_mgr.get_stop_lsn();
+
+  metadata_to_lsn = backup_ctxt.log_status.lsn_checkpoint;
```

Comment:
afaik,  this is the core change Instead of using a checkpiont_lsn by query (at alter point of time). we now use the checkpoint_lsn from the sync point (log_status). Is the understanding right?

Also, what will happen to the transactions that happened after checkpoint_lsn but before backup ended. With your approach, do we still copy the extra redo?

Reply:
yeah, The understanding is correct. 

metadata_last_lsn would be applied and then rollback if checkpoint LSN is less.

Reply:
can you please add testcase for this scenario and verify that rollback happens and the data we see is before binlog_pos + checkpoint_lsn?